packages/scene-composer/src/utils/entityModelUtils/SceneLifecycleModule.ts

```diff
@@ -0,0 +1,40 @@
+import { GetSceneCommandOutput } from '@aws-sdk/client-iottwinmaker';
+
+import {
+  LAYER_ROOT_ENTITY_ID,
+  LAYER_ROOT_ENTITY_NAME,
+  SCENE_ROOT_ENTITY_COMPONENT_NAME,
+  DYNAMIC_SCENE_CAPABILITY,
+} from '../../common/entityModelConstants';
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { KnownSceneProperty } from '../../interfaces';
+import { emptyScene } from '../emptyScene';
+
+import { updateSceneEntityComponent } from './sceneComponent';
+
+export class SceneLifecycleModule {
+  createDynamicScene = async (): Promise<GetSceneCommandOutput | Error> => {
+    try {
+      if (!getGlobalSettings().twinMakerSceneMetadataModule) {
+        return Error('SceneMetadataModule is undefined');
+      }
+      await getGlobalSettings().twinMakerSceneMetadataModule!.createScene({ capabilities: [DYNAMIC_SCENE_CAPABILITY] });
+      const sceneInfo = await getGlobalSettings().twinMakerSceneMetadataModule!.getSceneInfo();
+      const sceneRootEntityId = sceneInfo?.sceneMetadata?.[KnownSceneProperty.SceneRootEntityId];
+      const entityComponentModelDefault = updateSceneEntityComponent(emptyScene);
+      getGlobalSettings().twinMakerSceneMetadataModule!.updateSceneEntity({
+        entityId: sceneRootEntityId,
+        componentUpdates: {
+          [SCENE_ROOT_ENTITY_COMPONENT_NAME]: entityComponentModelDefault,
+        },
+      });
+      getGlobalSettings().twinMakerSceneMetadataModule!.createSceneEntity({
+        entityId: LAYER_ROOT_ENTITY_ID,
+        entityName: LAYER_ROOT_ENTITY_NAME,
+      });
+      return sceneInfo;
+    } catch (error) {
+      throw new Error('Error creating dynamic scene');
+    }
```

Comment:
this will convert all error into this generic error message and won't have much information. you don't need to try catch here and errors will just be thrown to caller to display

Reply:
Understood, done!